daemon/cluster/executor/container/adapter.go

```diff
@@ -65,17 +66,44 @@ func (c *containerAdapter) pullImage(ctx context.Context) error {
 	}()
 
 	dec := json.NewDecoder(pr)
+	dec.UseNumber()
 	m := map[string]interface{}{}
+	spamLimiter := rate.NewLimiter(rate.Every(1000*time.Millisecond), 1)
```

Comment:
@thaJeztah IMHO not a substantial enough improvement for me to spend an 30minutes - hour reworking it for percents instead of rates, but if you think it's important I am willing to do so